libs/configuration-variables-resolver/src/VariablesLoader/ConfigurationVariablesLoader.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ConfigurationVariablesResolver\VariablesLoader;
+
+use Keboola\ConfigurationVariablesResolver\ComponentsClientHelper;
+use Keboola\ConfigurationVariablesResolver\Exception\UserException;
+use Psr\Log\LoggerInterface;
+
+class ConfigurationVariablesLoader
+{
+    public function __construct(
+        private readonly ComponentsClientHelper $componentsHelper,
+        private readonly LoggerInterface $logger,
+    ) {
+    }
+
+    /**
+     * @param array{variables_id?: string|null, variables_values_id?: string|null} $configuration
+     * @param non-empty-string|null $variableValuesId
+     * @param array{values?: list<array{name: scalar, value: scalar}>|null}|null $variableValuesData
+     * @return array<non-empty-string, string>
+     */
+    public function loadVariables(array $configuration, ?string $variableValuesId, ?array $variableValuesData): array
+    {
+        if ($variableValuesId && !empty($variableValuesData['values'])) {
+            throw new UserException('Only one of variableValuesId and variableValuesData can be entered.');
+        }
+
+        $variablesId = $configuration['variables_id'] ?? null;
+        $variablesValuesId = $configuration['variables_values_id'] ?? null;
+
+        if (!$variablesId) {
+            return [];
+        }
+
+        $variablesConfiguration = $this->componentsHelper->getVariablesConfiguration($variablesId);
+        $variablesData = $this->loadVariablesData(
+            $variablesId,
+            $variablesValuesId,
+            $variableValuesId,
+            $variableValuesData,
+        );
+
+        $values = [];
+        foreach ($variablesData['values'] as $row) {
+            $key = (string) $row['name'];
+            $value = (string) $row['value'];
+
+            if ($key === '') {
+                throw new UserException('Variable name cannot be empty.');
+            }
+
+            $values[$key] = $value;
+        }
+
+        foreach ($variablesConfiguration['variables'] as $variable) {
+            if (!array_key_exists($variable['name'], $values)) {
+                throw new UserException(sprintf('No value provided for variable "%s".', $variable['name']));
+            }
+        }
```

Comment:
Je tohle k necemu dobry? Technicky je uplne jedno, jestli ta promenna je "zadefinovana" a je to jediny duvod, proc se vys dela call `$this->componentsHelper->getVariablesConfiguration($variablesId)`.

Reply:
Ja myslim, ze se to muze vyhodit